www/src/components/Menu.tsx

```diff
@@ -154,7 +173,7 @@ const foundationLinks = [
   'Colors', 'Elevation', 'Typography', 'Layout', 'Spacing', 'Icons', 'CSS-Utilities', 'Responsive', 'Brand-icons',
 ];
 
-function Menu() {
+function Menu({ componentName, componentCategories }) {
```

Comment:
I think you can simplify the code a lot if you just look at the current URL and identify which which menu item is active based on that instead of using `componentName` and `componentCategories`, e.g.:
```js
import { useLocation } from '@reach/router';
...
function Menu() {
  const {
    settings,
    handleSettingsChange,
  } = useContext(SettingsContext);
  const location = useLocation();
  const { components } = useStaticQuery(menuQuery);
  const { categories }: IMenuQueryComponents = components;

  ....

  // foundations section
  <Collapsible
    styling="basic"
    title="Foundations"
    defaultOpen={location.pathname.startsWith('/foundations')}
  >
    <ul className="list-unstyled foundations-list">
      {foundationLinks.map(link => (
        <li key={link}>
          <Link
            className={classNames({ active: location.pathname.endsWith(link.toLowerCase().trim()) })}
            to={`/foundations/${link.toLowerCase().trim()}`}
          >
            {link.replace(/-/g, ' ')}
          </Link>
        </li>
      ))}
    </ul>
  </Collapsible>

  ...

  // components section
  <MenuComponentList>
    {categories.map(({ fieldValue, nodes }) => (
      <Collapsible
        className="pgn-doc__menu-component--list-category"
        key={fieldValue}
        styling="basic"
        title={fieldValue}
        defaultOpen={nodes.some((node) => node.fields.slug === location.pathname)} // this is not very efficient but I think it's fine and will not cause performance issues
      >
        <ul className="list-unstyled">
          {nodes.map((node) => (
            <ComponentNavItem
              key={node.id}
              {...node}
              componentName={componentName}
              isActive={location.pathname === node.fields.slug}  // add this prop to ComponentNavItem instead of componentName and display active state based on it
            />
          ))}
        </ul>
      </Collapsible>
    ))}
  </MenuComponentList>
```

I think with this approach you will need to make changes only in this file and not touch any other

Reply:
Corrected, thanks!